CounterpointGenerator/CounterpointGenerator/Rules/startEndPerfectRule.cs

```diff
@@ -0,0 +1,17 @@
+namespace CounterpointGenerator{
+    public class startEndPerfectRule: IRules {
+        /**
+         * The first and last notes of the counterpoint must be a perfect consonance
+         */
+        
+        public List<int> apply(List<int> possibilities, int currentNote, int position, int length){
```

Comment:
Because of how these arguments vary, maybe you want to create a ruleInput interface with optional members?

Reply:
Still thinking about how I wanna properly tackle this. I did some light googling yesterday and Friday and interfaces with optional members seem to be very heavily frowned upon. Will be resolved in a new issue + PR

Reply:
[Possible idea for implementation?](https://softwareengineering.stackexchange.com/a/345046)

Reply:
Resolved in a9420f1